AudioDeviceExample/ViewController.swift

```diff
@@ -0,0 +1,465 @@
+        if (newCollection.horizontalSizeClass == .regular ||
+            (newCollection.horizontalSizeClass == .compact && newCollection.verticalSizeClass == .compact)) {
+            remoteViewStack.axis = .horizontal
+        } else {
+            remoteViewStack.axis = .vertical
+        }
+    }
+
+    // Update our UI based upon if we are in a Room or not
+    func showRoomUI(inRoom: Bool) {
+        self.connectButton.isHidden = inRoom
+        self.connectButton.isEnabled = !inRoom
+        self.roomTextField.isHidden = inRoom
+        self.roomLine.isHidden = inRoom
+        self.roomLabel.isHidden = inRoom
+        self.disconnectButton.isHidden = !inRoom
+        self.disconnectButton.isEnabled = inRoom
+        UIApplication.shared.isIdleTimerDisabled = inRoom
+        if #available(iOS 11.0, *) {
+            self.setNeedsUpdateOfHomeIndicatorAutoHidden()
+        }
+        self.setNeedsStatusBarAppearanceUpdate()
+
+        self.navigationController?.setNavigationBarHidden(inRoom, animated: true)
+    }
+
+    func dismissKeyboard() {
+        if (self.roomTextField.isFirstResponder) {
+            self.roomTextField.resignFirstResponder()
+        }
+    }
+
+    func logMessage(messageText: String) {
+        messageLabel.text = messageText
+
+        if (messageLabel.alpha < 1.0) {
+            self.messageLabel.isHidden = false
+            UIView.animate(withDuration: 0.4, animations: {
+                self.messageLabel.alpha = 1.0
+            })
+        }
+
+        // Hide the message with a delay.
+        self.messageTimer?.invalidate()
+
+        self.messageTimer = Timer.init(timeInterval: TimeInterval(6),
+                                       target: self,
+                                       selector: #selector(hideMessageLabel),
+                                       userInfo: nil,
+                                       repeats: false)
+        RunLoop.main.add(self.messageTimer, forMode: .commonModes)
+    }
+
+    @objc func hideMessageLabel() {
+        if (self.messageLabel.isHidden == false) {
+            UIView.animate(withDuration: 0.6, animations: {
+                self.messageLabel.alpha = 0
+            }, completion: { (complete) in
+                if (complete) {
+                    self.messageLabel.isHidden = true
+                }
+            })
+        }
+    }
+
+    func prepareLocalMedia() {
+        /*
+         * The important thing to remember when using a custom TVIAudioDevice is that the device must be set
+         * before performing any other actions with the SDK (such as creating Tracks, or connecting to a Room).
+         */
+        TwilioVideo.audioDevice = ExampleCoreAudioDevice.init();
+
+        /*
+         * ExampleCoreAudioDevice is a playback only device. Because of this, any attempts to create a
+         * TVILocalAudioTrack will result in an exception being thrown, and nil being returned. In this example we will
```

Comment:
Are we 100% sure if nil will be returned. . should remove it if we are not.

Reply:
This comment is describing the intent of the APIs. I think its important to mention this behavior, despite the fact that we have a small race around this in beta1.